sparse/core.py

```diff
@@ -611,34 +619,46 @@ def elemwise_binary(self, func, other, *args, **kwargs):
 
         # Sort self.coords in lexographical order using record arrays
         self_coords = np.rec.fromarrays(self.coords)
-        i = np.argsort(self_coords)
-        self_coords = self_coords[i]
-        self_data = self.data[i]
+        self_sort_order = np.argsort(self_coords)
+        self_coords = self_coords[self_sort_order]
+        self_data = self.data[self_sort_order]
 
-        # Convert other.coords to a record array
+        # Sort other.coords in lexographical order using record arrays
         other_coords = np.rec.fromarrays(other.coords)
-        other_data = other.data
+        other_sort_order = np.argsort(other_coords)
+        other_coords = other_coords[other_sort_order]
+        other_data = other.data[other_sort_order]
 
         # Find matches between self.coords and other.coords
-        j = np.searchsorted(self_coords, other_coords)
-        if len(self_coords):
-            matched_other = (other_coords == self_coords[j % len(self_coords)])
-        else:
-            matched_other = np.zeros(shape=(0,), dtype=bool)
-        matched_self = j[matched_other]
+        # TODO: implement a modified exponential search to find next fitting coordinate faster
+        matched_self = np.zeros(shape=(len(self_coords),), dtype=bool)
+        matched_other = np.zeros(shape=(len(other_coords),), dtype=bool)
+        i = 0
+        j = 0
+        while (i < len(self_coords)) & (j < len(other_coords)):
+            self_coord = tuple(self_coords[i])
+            other_coord = tuple(other_coords[j])
+            if self_coord == other_coord:
+                matched_self[i] = True
+                matched_other[j] = True
+                i += 1
+                j += 1
+            elif self_coord < other_coord:
+                i += 1
+            else:
+                j += 1
```

Comment:
Surly it is not optimal in terms of performance. But the performance should be about the same as running `np.rec.fromarrays(self.coords)` plus `np.rec.fromarrays(other.coords)`. So in the worst case it may affects the performance by about a constant factor 2. 
Generally: his was more a fast approach to fix #25. This part of the code is for my own project quite performance critical. So I will rewrite this section about next week with focused on asymptotic runtime and memory consumption.